src/main/java/com/incentives/piggyback/invoice/serviceImpl/InvoiceServiceImpl.java

```diff
@@ -0,0 +1,30 @@
+package com.incentives.piggyback.invoice.serviceImpl;
+
+import com.incentives.piggyback.invoice.exception.InvoiceNotFoundException;
+import com.incentives.piggyback.invoice.model.Invoice;
+import com.incentives.piggyback.invoice.repository.InvoiceServiceRepository;
+import com.incentives.piggyback.invoice.service.InvoiceService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+@Service
+public class InvoiceServiceImpl implements InvoiceService {
+
+    @Autowired
+    private InvoiceServiceRepository invoiceServiceRepository;
+
+    @Override
+    public ResponseEntity<Invoice> getInvoiceById(Long id) {
+        return ResponseEntity.ok(invoiceServiceRepository.findById(id).orElseThrow(()->new InvoiceNotFoundException(id)));
+}
+
+    @Override
+    public ResponseEntity<Invoice> payInvoiceById(Long id, Invoice invoice) {
+        invoiceServiceRepository.findById(id).orElseThrow(()->new InvoiceNotFoundException(id));
+        Invoice updatedInvoice = invoiceServiceRepository.save(invoice);
```

Comment:
Instead of directly saving what client is sending, you can update only not null fields that should be allowed. If client sends lesser info, that will be saved.